libmeepgeom/meepgeom.cpp

```diff
@@ -1464,7 +1464,8 @@ void add_absorbing_layer(absorber_list alist,
 /***************************************************************/
 /***************************************************************/
 /***************************************************************/
-void set_materials_from_geometry(meep::structure *s,
+void set_materials_from_geometry(vector3 size,
+                                 meep::structure *s,
```

Comment:
Can't you get `size` from `s->user_volume` (the grid `volume` before symmetry reduction) or `s->v` (the corresponding `volume`)?

Reply:
`s->user_volume` works. Thanks.  This is green now and should be ready to merge.